src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts

```diff
@@ -0,0 +1,240 @@
+  daemons: RgwDaemon[] = [];
+  selectedCluster = '';
+  clusterDetailsArray: any;
+  isMultiCluster = false;
+  exportTokenForm: CdFormGroup;
+  realms: any;
+  loading = false;
+  icons = Icons;
+  rgwEndpoints: { value: any[]; options: any[]; messages: any };
+
+  constructor(
+    private wizardStepsService: WizardStepsService,
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    private rgwDaemonService: RgwDaemonService,
+    private multiClusterService: MultiClusterService,
+    private rgwMultisiteService: RgwMultisiteService,
+    public notificationService: NotificationService,
+    private router: Router
+  ) {
+    this.currentStepSub = this.wizardStepsService
+      .getCurrentStep()
+      .subscribe((step: WizardStepModel) => {
+        this.currentStep = step;
+      });
+    this.currentStep.stepIndex = 1;
+    this.createForm();
+    this.rgwEndpoints = {
+      value: [],
+      options: [],
+      messages: new SelectMessages({
+        empty: $localize`There are no endpoints.`,
+        filter: $localize`Add endpoints`
+      })
+    };
+  }
+
+  ngOnInit(): void {
+    this.rgwDaemonService.list().subscribe((daemons) => {
+      this.daemons = daemons;
+      this.daemons.forEach((daemon) => {
+        this.rgwEndpoints.value.push('http://' + daemon.server_hostname + ':' + daemon.port);
+
+        const options: SelectOption[] = [];
+        _.forEach(this.rgwEndpoints.value, (endpoint: string) => {
+          const option = new SelectOption(false, endpoint, '');
+          options.push(option);
+        });
+        this.rgwEndpoints.options = [...options];
+      });
+    });
+
+    this.multiClusterService.getCluster().subscribe((clusters) => {
+      this.clusterDetailsArray = Object.values(clusters['config'])
+        .flat()
+        .filter((cluster) => cluster['cluster_alias'] !== 'local-cluster');
+      this.isMultiCluster = this.clusterDetailsArray.length > 0;
+      if (!this.isMultiCluster) {
+        this.stepTitles = ['Create Realm & Zonegroup', 'Create Zone', 'Export Multi-Site token'];
+      } else {
+        this.selectedCluster = this.clusterDetailsArray[0]['name'];
+      }
+    });
+  }
+
+  createForm() {
+    this.multisiteSetupForm = new CdFormGroup({
+      realmName: new UntypedFormControl('default_realm', {
+        validators: [Validators.required]
+      }),
+      zonegroupName: new UntypedFormControl('default_zonegroup', {
+        validators: [Validators.required]
+      }),
+      zonegroup_endpoints: new UntypedFormControl(null, [Validators.required]),
+      zoneName: new UntypedFormControl('default_zone', {
+        validators: [Validators.required]
+      }),
+      zone_endpoints: new UntypedFormControl(null, {
+        validators: [Validators.required]
+      }),
+      username: new UntypedFormControl('default_system_user', {
+        validators: [Validators.required]
+      }),
+      cluster: new UntypedFormControl(null, {
```

Comment:
Since this fields are required to be unique, should we check if `default_realm` `default_zonegroup`.... are already in place and act accordingly?

Reply:
The wizard is only accessible when we have just the default configuration in place i.e `default` zonegroup and `default` zone